stopify-continuations/src/runtime/lazyRuntime.ts

```diff
@@ -13,9 +13,15 @@ export class LazyRuntime extends common.Runtime {
 
   makeCont(stack: common.Stack) {
     const savedDelimitDepth = this.delimitDepth;
-    return (v: any) => {
+
+    return (v: any, err: any=this.noErrorProvided) => {
+      const throwExn = err !== this.noErrorProvided;
       this.delimitDepth = savedDelimitDepth;
-      throw new common.Restore([this.topK(() => v), ...stack]);
+      let restarter = () => {
+        if(throwExn) { throw err; }
+        else { return v; }
+      }
+      throw new common.Restore([this.topK(restarter), ...stack]);
```

Comment:
@jpolitz For shallow runtimes, the last stack frame to the invoked is `$__R.stack[0]`. Since deepRuntimes invoke the `$__R.stack[$__R.stack.length - 1]` and start running the computation, should the top frame be placed at the end of the stack?

Reply:
I don't know – I would be searching Slack history and old commits to try and find it. I remember trying both and one worked.

Reply:
I think I have it figured out.